File: python/isbn-verifier/isbn_verifier.py
def is_valid(isbn):
    sum = 0
    no_dash_isbn = isbn.lower().replace("-","")
    if len(no_dash_isbn) != 10:
        return False
    for i,v in enumerate(no_dash_isbn):
        if v == "x" and i == 9:
            v= 10
        elif not v.isdigit():
            return False
        sum += int(v)*(10-i)
    return sum % 11 == 0

File: go/isbn-verifier/isbn_verifier.go
package isbn

import (
	"strconv"
	"strings"
)

func isdigit(r rune) bool {
	return (r >= '0' && r <= '9')
}
func IsValidISBN(isbn string) bool {
	isbn = strings.ReplaceAll(isbn, "-", "")
	if len(isbn) != 10 {
		return false
	}
	for i, c := range isbn {
		if !isdigit(c) {
		//Or just unicode.IsDidit :|  
		//I've reinvented the wheel

			if i == 9 && c == 'X' {
				continue
			} else {
				return false
			}
		}
	}

	var sum, multiplier int
	for i := 0; i < 10; i++ {
		if i == 9 && string(isbn[i]) == "X" {
			multiplier = 10
		} else {
			multiplier, _ = strconv.Atoi(string(isbn[i]))
		}
		sum += multiplier * (10 - i)
	}
	if sum%11 == 0 {
		return true
	} else {
		return false
	}
}
